src/executor/current_thread.rs

```diff
@@ -0,0 +1,361 @@
+//! Execute tasks on the current thread
+
+use Async;
+use executor::{self, Spawn};
+use future::{self, Future, Executor, ExecuteError, ExecuteErrorKind};
+use scheduler;
+use task_impl::ThreadNotify;
+
+use std::prelude::v1::*;
+
+use std::{fmt, ptr, thread};
+use std::cell::Cell;
+use std::rc::Rc;
+use std::sync::Arc;
+
+/// Executes tasks on the current thread.
+#[derive(Debug, Clone)]
+pub struct CurrentThread {
+    // Prevent the handle from moving across threads.
+    _p: ::std::marker::PhantomData<Rc<()>>,
+}
+
+/// Executes dameonized tasks on the current thread.
```

Comment:
Here and elsewhere it would be good to clarify what it means for a task to be a "daemon"-- i.e. it executes indefinitely, and it does not prevent functions like block_on_all from returning.